src/libraries/System.Text.Json/tests/System.Text.Json.SourceGeneration.Tests/roslyn4.4.rd.xml

```diff
@@ -0,0 +1,56 @@
+<Directives>
+  <Application>
+    <!-- NB Shared specializations must also be replicated in the roslyn3.11.rd.xml file -->
+    <Assembly Name="System.Text.Json.SourceGeneration.Roslyn4.4.Tests">
```

Comment:
why are those needed? Do we know the reason they're trimmed out?

Reply:
They are helping the AOT compiler generate generic method specializations we know will be needed at runtime (but can't be inferred at compile time).

Reply:
do we know why they can't be inferred?

Reply:
It's the usual issue of trying to do `Type.MakeGenericType`/`Method.MakeGenericMethod` in AOT. We're working on a fix specifically for theories though: https://github.com/dotnet/runtime/pull/86975#discussion_r1212381894